### DISCUSSION TITLE
Easily detect if a speaker is connected to the TV

### DISCUSSION THREAD

**Discussion by danmcpherson** [avantrec/soco-cli#46] [Ideas]
First, let me say thank you for this software. It's exactly what I had been missing, and it works brilliantly. 

One thing I would like to see is an easy way to determine if a Speaker is connected to the TV. I see that this is exposed via the "info" action, but it's a bit of a challenge to easily use in other integrations. It would be great if it was exposed much like the "status_light". 

Thanks again, really appreciate your work.

**Comment by pwt** [comment]
Hi Dan:

The `info` action reports `is_playing_tv = False` as one of its line items, which I assume tells one whether the TV source is actively playing or not, and not whether there's anything connected to the to the TV input. Is that sufficient?

If so, I can pull that out into a separate action: `sonos speaker playing_tv`, returning `true` or `false`, if that helps?

I don't currently have any Sonos devices connected to a TV, so I'd have to depend on others for testing.

**Comment by danmcpherson** [comment]
Thanks for the reply. That value looks to be exactly what I’m looking for. Breaking it out would be perfect as some integrations I’m looking to do make it very challenging to parse the larger response payload.

More than happy to test it.

**Comment by pwt** [comment]
I've released v0.4.55 that adds the `playing_tv` (or `is_playing_tv`) action, which reports `yes` or `no`. Let me know if you encounter any issues.

**Reply by danmcpherson** [reply]
That was crazy quick. I will give it a test tomorrow and let you know. Really appreciate it, can I support the project in some way?

**Reply by pwt** [reply]
It's a very small addition, so didn't take long.
I'm just happy if folks get value out of the project, and even happier if they tell me so :-)

**Reply by danmcpherson** [reply]
Works perfectly. Thanks very much. I've now got a way to automatically put the window blinds down in the TV room that doesn't require me polling for the appearance of a Chromecast (that is plugged into the TV) appearing on the network!
